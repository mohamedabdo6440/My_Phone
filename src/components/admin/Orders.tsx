// local modules
import AdminLayout from '@/components/admin/AdminLayout'
import DataGrid, { DataRowDef } from '@/components/admin/DataGrid'
import { NextPageWithLayout } from '@/pages/_app'
import styles from '@/styles/components/admin/orders.module.scss'
import { FaRegTrashAlt } from 'react-icons/fa'
import { MdAdd, MdEdit, MdSearch } from 'react-icons/md'

interface OrdersProps {
    title: string
}

const Orders: NextPageWithLayout<OrdersProps> = ({ title }) => {
    return (
        <div className={styles.container}>
            <div className={styles.header}>
                <h2>{title}</h2>
                <button>
                    <MdAdd />
                    <span>Add new</span>
                </button>
            </div>
            <form className={styles.form}>
                <input type='text' placeholder='Order Id' />
                <input type='text' placeholder='Customer Name' />
                <div className={styles.date}>
                    <label htmlFor='fromDate'>From Date</label>
                    <input type='date' placeholder='From Date' />
                </div>
                <div className={styles.date}>
                    <label htmlFor='toDate'>To Date</label>
                    <input type='date' placeholder='To Date' />
                </div>
                <select name='status'>
                    <option value=''>Status</option>
                    <option value=''>Awaiting</option>
                    <option value=''>Completed</option>
                    <option value=''>Archived</option>
                    <option value=''>Abandoned</option>
                </select>
                <button>
                    <span>Search</span>
                    <MdSearch />
                </button>
            </form>

            <DataGrid checkboxSelection columns={columns} rows={rows} />
        </div>
    )
}

const Actions = () => {
    return (
        <div className={styles.actions}>
            <MdEdit className={styles.edit} />
            <FaRegTrashAlt className={styles.trash} />
        </div>
    )
}

const columns = [
    {
        field: 'id' as const,
        headerName: 'Order ID',
    },
    {
        field: 'customer' as const,
        headerName: 'Customer',
    },
    {
        field: 'date' as const,
        headerName: 'Date',
    },
    {
        field: 'qty' as const,
        headerName: 'QTY',
    },
    {
        field: 'price' as const,
        headerName: 'Price',
    },
    {
        field: 'paymentMethod' as const,
        headerName: 'Payment Method',
    },
    {
        field: 'status' as const,
        headerName: 'Status',
    },
    {
        field: 'actions' as const,
        headerName: 'Actions',
    },
]

const rows: DataRowDef<typeof columns> = [
    {
        status: 'completed',
        id: 'qrlvlqc4',
        customer: 'JJ Brown',
        date: 'Oct 20, 2022',
        price: 560,
        paymentMethod: 'Cash',
        actions: <Actions />,
        qty: 2,
    },
    {
        id: 'qrlvlqc4',
        customer: 'JJ Brown',
        date: 'Oct 20, 2022',
        qty: 2,
        price: 560,
        paymentMethod: 'Cash',
        status: 'completed',
        actions: <Actions />,
    },
]

Orders.getLayout = function getLayout(page) {
    return <AdminLayout>{page}</AdminLayout>
}

export default Orders
