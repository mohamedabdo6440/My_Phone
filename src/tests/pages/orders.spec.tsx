// @ts-ignore
import React from "react"
import "@testing-library/jest-dom"
import {act, render} from "@testing-library/react";
import Orders from "@/pages/orders";
import {Provider} from "react-redux";
import configureStore from 'redux-mock-store'

test("type buy status all", async () => {
    const store = configureStore()
    const state = {
        theme: "light",
        order: {
            tab: "buy",
            filter: "all",
        }
    }
    const res = await act(async () => {
        return render(
            <Provider store={store(state)}>
                <Orders status={200} message={"ok"}/>
            </Provider>
        )
    })
    expect(res.baseElement).toMatchSnapshot()
})

test("type sell status pending", async () => {
    const store = configureStore()
    const state = {
        theme: "light",
        order: {
            tab: "sell",
            filter: "pending",
        }
    }
    const res = await act(async () => {
        return render(
            <Provider store={store(state)}>
                <Orders status={200} message={"ok"}/>
            </Provider>
        )
    })
    expect(res.baseElement).toMatchSnapshot()
})
