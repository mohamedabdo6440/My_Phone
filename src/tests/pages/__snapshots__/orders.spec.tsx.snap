// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`type buy status all 1`] = `
<body>
  <div>
    <section
      class="undefined undefined"
    >
      <h2>
        My Orders
      </h2>
      <div>
        <h3
          class="null"
        >
          Sell Smart
        </h3>
        <h3
          class="undefined"
        >
          Buy Smart
        </h3>
        <h3
          class="null"
        >
          Repair Smart
        </h3>
      </div>
      <div>
        <div>
          <span
            class="undefined"
          >
            All Orders
          </span>
          <span
            class="null"
          >
            Pending Orders
          </span>
          <span
            class="null"
          >
            Completed Orders
          </span>
        </div>
        <table>
          <thead>
            <tr>
              <th>
                Order Date
              </th>
              <th>
                Carrier
              </th>
              <th>
                Order Value
              </th>
              <th>
                Order Status
              </th>
              <th>
                Operation
              </th>
            </tr>
          </thead>
          <tbody>
            <tr>
              <td>
                Wed, 01 Jan 2020 00:00:00 GMT
              </td>
              <td>
                <div>
                  <span
                    style="box-sizing: border-box; display: inline-block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: relative; max-width: 100%;"
                  >
                    <span
                      style="box-sizing: border-box; display: block; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; max-width: 100%;"
                    >
                      <img
                        alt=""
                        aria-hidden="true"
                        src="data:image/svg+xml,%3csvg%20xmlns=%27http://www.w3.org/2000/svg%27%20version=%271.1%27%20width=%271%27%20height=%271%27/%3e"
                        style="display: block; max-width: 100%; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px;"
                      />
                    </span>
                    <img
                      alt="Sprint"
                      data-nimg="intrinsic"
                      decoding="async"
                      src=""
                      style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
                    />
                    <noscript />
                  </span>
                </div>
              </td>
              <td>
                $
                20
              </td>
              <td>
                $
                pending
              </td>
              <td>
                <button>
                  Details
                </button>
              </td>
            </tr>
            <tr>
              <td>
                Wed, 01 Jan 2020 00:00:00 GMT
              </td>
              <td>
                <div>
                  <span
                    style="box-sizing: border-box; display: inline-block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: relative; max-width: 100%;"
                  >
                    <span
                      style="box-sizing: border-box; display: block; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; max-width: 100%;"
                    >
                      <img
                        alt=""
                        aria-hidden="true"
                        src="data:image/svg+xml,%3csvg%20xmlns=%27http://www.w3.org/2000/svg%27%20version=%271.1%27%20width=%271%27%20height=%271%27/%3e"
                        style="display: block; max-width: 100%; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px;"
                      />
                    </span>
                    <img
                      alt="Sprint"
                      data-nimg="intrinsic"
                      decoding="async"
                      src=""
                      style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
                    />
                    <noscript />
                  </span>
                </div>
              </td>
              <td>
                $
                34
              </td>
              <td>
                $
                completed
              </td>
              <td>
                <button>
                  Details
                </button>
              </td>
            </tr>
          </tbody>
        </table>
      </div>
    </section>
  </div>
</body>
`;

exports[`type sell status pending 1`] = `
<body>
  <div>
    <section
      class="undefined undefined"
    >
      <h2>
        My Orders
      </h2>
      <div>
        <h3
          class="undefined"
        >
          Sell Smart
        </h3>
        <h3
          class="null"
        >
          Buy Smart
        </h3>
        <h3
          class="null"
        >
          Repair Smart
        </h3>
      </div>
      <div>
        <div>
          <span
            class="null"
          >
            All Orders
          </span>
          <span
            class="undefined"
          >
            Pending Orders
          </span>
          <span
            class="null"
          >
            Completed Orders
          </span>
        </div>
        <table>
          <thead>
            <tr>
              <th>
                Order Date
              </th>
              <th>
                Carrier
              </th>
              <th>
                Order Value
              </th>
              <th>
                Order Status
              </th>
              <th>
                Operation
              </th>
            </tr>
          </thead>
          <tbody>
            <tr>
              <td>
                Wed, 01 Jan 2020 00:00:00 GMT
              </td>
              <td>
                <div>
                  <span
                    style="box-sizing: border-box; display: inline-block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: relative; max-width: 100%;"
                  >
                    <span
                      style="box-sizing: border-box; display: block; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; max-width: 100%;"
                    >
                      <img
                        alt=""
                        aria-hidden="true"
                        src="data:image/svg+xml,%3csvg%20xmlns=%27http://www.w3.org/2000/svg%27%20version=%271.1%27%20width=%271%27%20height=%271%27/%3e"
                        style="display: block; max-width: 100%; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px;"
                      />
                    </span>
                    <img
                      alt="Sprint"
                      data-nimg="intrinsic"
                      decoding="async"
                      src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
                      style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
                    />
                    <noscript />
                  </span>
                </div>
              </td>
              <td>
                $
                200
              </td>
              <td>
                $
                pending
              </td>
              <td>
                <button>
                  Details
                </button>
              </td>
            </tr>
            <tr>
              <td>
                Wed, 01 Jan 2020 00:00:00 GMT
              </td>
              <td>
                <div>
                  <span
                    style="box-sizing: border-box; display: inline-block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: relative; max-width: 100%;"
                  >
                    <span
                      style="box-sizing: border-box; display: block; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; max-width: 100%;"
                    >
                      <img
                        alt=""
                        aria-hidden="true"
                        src="data:image/svg+xml,%3csvg%20xmlns=%27http://www.w3.org/2000/svg%27%20version=%271.1%27%20width=%271%27%20height=%271%27/%3e"
                        style="display: block; max-width: 100%; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px;"
                      />
                    </span>
                    <img
                      alt="Sprint"
                      data-nimg="intrinsic"
                      decoding="async"
                      src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
                      style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
                    />
                    <noscript />
                  </span>
                </div>
              </td>
              <td>
                $
                2000
              </td>
              <td>
                $
                pending
              </td>
              <td>
                <button>
                  Details
                </button>
              </td>
            </tr>
          </tbody>
        </table>
      </div>
    </section>
  </div>
</body>
`;
